modules/images/image-loading-optimization/rest-api.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * REST API integration for the module.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Register endpoint for storage of metrics.
+ */
+function image_loading_optimization_register_endpoint() {
+
+	$dom_rect_schema = array(
+		'type'       => 'object',
+		'properties' => array(
+			'width'  => array(
+				'type'    => 'number',
+				'minimum' => 0,
+			),
+			'height' => array(
+				'type'    => 'number',
+				'minimum' => 0,
+			),
+			// TODO: There are other properties to define if we need them: x, y, top, right, bottom, left.
+		),
+	);
+
+	register_rest_route(
+		'perflab/v1',
+		'/image-loading-optimization/metrics-storage',
+		array(
+			'methods'             => 'POST',
+			'callback'            => 'image_loading_optimization_handle_rest_request',
+			'permission_callback' => '__return_true', // Needs to be available to unauthenticated visitors.
+			'args'                => array(
```

Comment:
Is the URL also passed as an argument?

Reply:
Good point! 😊  However, this could also obtained via the `Referer` HTTP request header. But probably better to be explicit, even though perhaps `Referer` would be a bit safer to rely on. If nothing else, a custom validation callback could be used for the `url` arg to make sure it matches the `Referer`.

Reply:
Added `url` in b4e29d609647c8e23fc4db6532e50ba8692e82c5